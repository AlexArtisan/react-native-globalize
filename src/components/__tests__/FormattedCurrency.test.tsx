/*!
 * React Native Globalize
 *
 * Copyright 2015-2020 Josh Swan
 * Released under the MIT license
 * https://github.com/joshswan/react-native-globalize/blob/master/LICENSE
 */

import React from 'react';
import { createWithGlobalize } from '../../../test/createWithGlobalize';
import { FormattedCurrency } from '..';

describe('<FormattedCurrency />', () => {
  test('renders correctly', () => {
    const tree = createWithGlobalize(<FormattedCurrency value={10} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('symbolForm', () => {
    test('supports symbolForm prop to use alt narrow currency symbol', () => {
      const tree = createWithGlobalize(
        <FormattedCurrency currency="CAD" symbolForm="narrow" value={10} />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
