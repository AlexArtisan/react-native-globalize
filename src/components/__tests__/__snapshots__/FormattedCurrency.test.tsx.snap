// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedCurrency /> renders correctly 1`] = `
<Text
  accessibilityLabel=""
  accessible={true}
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
>
  $10.00
</Text>
`;

exports[`<FormattedCurrency /> symbolForm supports symbolForm prop to use alt narrow currency symbol 1`] = `
<Text
  accessibilityLabel=""
  accessible={true}
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
>
  $10.00
</Text>
`;
